contracts/GoodGhosting.sol

```diff
@@ -229,6 +231,9 @@ contract GoodGhosting is Ownable, Pausable {
         require(!player.withdrawn, "Player has already withdrawn");
         player.withdrawn = true;
         activePlayersCount = activePlayersCount.sub(1);
+        if (winnerCount > 0) {
+            winnerCount --;
+        }
```

Comment:
This would decrease the winner count at any segment, for any player - not only winners.
We prob first need to figure out if the player withdrawing is marked as a winner, and only then decrease the counter.

Reply:
@fodisi winnerCount is only incrememted in the last segment so that account is a winner
You can have a look at the the transfer dai code the increment happens there Only

Reply:
The issue with this is that we can be in the last segment, and then a non-winner decides to withdraw.
This scenario would result in the `winnerCounter` being decreased because of a non-winner player, which is not correct.

A second note, let's use SafeMath for subtractions just for keep the standard.

Reply:
and we don't need a for loop to check if the player was a winner or not. The player struct holds the `mostRecentSegmentPaid`, which can be used to compare if it's >= to `segmentCount-1`.

Reply:
oh I missed this case @fodisi good catch yaah besides winnerCounter > 0 there needs to be a last segment paid check too will add this with test coverage